0-bootstrap/modules/jenkins/README.md

```diff
@@ -0,0 +1,134 @@
+| folder\_id | The ID of a folder to host this project | string | `""` | no |
+| group\_org\_admins | Google Group for GCP Organization Administrators | string | n/a | yes |
+| org\_id | GCP Organization ID | string | n/a | yes |
+| project\_labels | Labels to apply to the project. | map(string) | `<map>` | no |
+| project\_prefix | Name prefix to use for projects created. | string | `"cft"` | no |
+| sa\_enable\_impersonation | Allow org_admins group to impersonate service account & enable APIs required. | bool | `"false"` | no |
+| storage\_bucket\_labels | Labels to apply to the storage bucket. | map(string) | `<map>` | no |
+| terraform\_sa\_email | Email for terraform service account. | string | n/a | yes |
+| terraform\_sa\_name | Fully-qualified name of the terraform service account. | string | n/a | yes |
+| terraform\_state\_bucket | Default state bucket, used in Cloud Build substitutions. | string | n/a | yes |
+| terraform\_version | Default terraform version. | string | `"0.12.24"` | no |
+| terraform\_version\_sha256sum | sha256sum for default terraform version. | string | `"602d2529aafdaa0f605c06adb7c72cfb585d8aa19b3f4d8d189b42589e27bf11"` | no |
+
+## Outputs
+
+| Name | Description |
+|------|-------------|
+| jenkins\_project\_id | Project where Jenkins Agent and terraform container image will reside. |
+| TODO csr\_repos | List of Cloud Source Repos created by the module, linked to Cloud Build triggers. |
+| TODO gcs\_bucket\_jenkins\_artifacts | Bucket used to store Jenkins artifacts in CICD project. |
+| TODO kms\_crypto\_key | KMS key created by the module. |
+| TODO kms\_keyring | KMS Keyring created by the module. |
+
+<!-- END OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
+
+## Requirements
+
+### Software
+
+-   [gcloud sdk](https://cloud.google.com/sdk/install) >= 206.0.0
+-   [Terraform](https://www.terraform.io/downloads.html) >= 0.12.6
+-   [terraform-provider-google] plugin 2.1.x
+-   [terraform-provider-google-beta] plugin 2.1.x
+
+### Permissions
+
+- `roles/billing.user` on supplied billing account
+- `roles/resourcemanager.organizationAdmin` on GCP Organization
+- `roles/resourcemanager.projectCreator` on GCP Organization or folder
+
+### APIs
+
+A project with the following APIs enabled must be used to host the
+resources of this module:
+
+- Google Cloud Resource Manager API: `cloudresourcemanager.googleapis.com`
+- Google Cloud Billing API: `cloudbilling.googleapis.com`
+- Google Cloud IAM API: `iam.googleapis.com`
+- Google Cloud Storage API `storage-api.googleapis.com`
+- Google Cloud Service Usage API: `serviceusage.googleapis.com`
+- Google Cloud Compute API: `compute.googleapis.com`
+- Google Cloud KMS API: `cloudkms.googleapis.com`
+
+This API can be enabled in the default project created during establishing an organization.
+
+### Troubleshooting
+
+When running `terraform apply` for the first time, you might see an message like this in the logs:
```

Comment:
I am not sure if this is necessary, if it is we can just link to project factory [troubleshooting](https://github.com/terraform-google-modules/terraform-google-project-factory/blob/master/docs/TROUBLESHOOTING.md) which has the same information. I am also thinking this can be part of general troubleshooting for this solution since there are quite a few moving pieces.

Reply:
Thank you! - Removed.